pre_commit/languages/pygrep.py

```diff
@@ -80,6 +81,8 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
             retv |= _process_filename_at_once(pattern, filename)
         else:
             retv |= _process_filename_by_line(pattern, filename)
+    if args.negate:
+        retv = 1 - retv
```

Comment:
this is buggy, when processing multiple files if any of them match it will return 0

Reply:
also `1 - retv` is much harder to understand than `retv = not retv`

Reply:
yes, you're right, thanks, sorry for having submitted this without having though it through